### PR TITLE
Courses

### PR DESCRIPTION
**Pull request by nheyek**
Implement basic functionality of courses.
Users can view only questions corresponding to their course.
Teachers create questions that correspond to their course.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java [hunk 1]
```diff
@@ -44,12 +44,6 @@
 import static ch.epfl.sweng.studyup.utils.Utils.XP_STEP;
 
 public class MainActivity extends NavigationStudent {
-    private static final CircularProgressIndicator.ProgressTextAdapter LEVEL_PROGRESS_TEXT = new CircularProgressIndicator.ProgressTextAdapter() {
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java]: Why did you remove this?

Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java]: Why did you delete this?

File: app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java [hunk 2]
```diff
@@ -83,7 +77,10 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_main);
 
         displayLoginSuccessMessage(getIntent());
-        Firestore.get().loadQuestions(this);
+
+        Firestore.loadQuestions(this);
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java]: Why not Firestore.get() there?

File: app/src/main/java/ch/epfl/sweng/studyup/course/Course.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package ch.epfl.sweng.studyup.course;
+
+import java.util.UUID;
+
+public class Course {
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/course/Course.java]: Why is this not an enum? It would make much more sense, courses are predefined and unique.

File: app/src/main/java/ch/epfl/sweng/studyup/firebase/Firestore.java [hunk 4]
```diff
@@ -256,18 +259,39 @@ public void onComplete(@NonNull Task<Void> task) {
...
+                    @Override
+                    public void onSuccess(DocumentSnapshot document) {
+                        if (document.exists()) {
+                            String courseId = null;
+                            if (Utils.isMockEnabled) {
+                                courseId = DEFAULT_COURSE_ID;
+                            }
+                            else {
+                                if (document.get(FB_COURSE_ID) != null) {
+                                    courseId = document.get(FB_COURSE_ID).toString();
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/firebase/Firestore.java]: Why only one course?

File: app/src/main/java/ch/epfl/sweng/studyup/firebase/Firestore.java [hunk 5]
```diff
@@ -295,25 +320,27 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
                         boolean getThatQuestion;
                         if(Player.get().getRole()) {
                             getThatQuestion = Player.get().getSciper() == QuestionAuthor;
+                            System.out.println("Role of teacher");
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/firebase/Firestore.java]: Use logs, they are much more convenient for debugging than prints...

File: app/src/main/java/ch/epfl/sweng/studyup/player/Player.java [hunk 6]
```diff
@@ -55,6 +56,7 @@
     private String username;
     private boolean isTeacher;
     private int sciper;
+    private String courseId;
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/player/Player.java]: Why does he have only one course? It doesn't really make sense.

File: app/src/main/java/ch/epfl/sweng/studyup/utils/Utils.java [hunk 7]
```diff
@@ -71,6 +73,7 @@
     public static final String GALLERY = "Gallery";
     public static final String CANCEL = "Cancel";
     public static final String JUSTONCE = "JUST ONCE";
+    public static final String DEFAULT_COURSE_ID = "bVW3QC2SMgLjVRTPLWld";
```
Comment (Gorzen) [inline app/src/main/java/ch/epfl/sweng/studyup/utils/Utils.java]: Why have something so nasty, why do you need a courses collection on firebase? Why not just put a list of courses in the player collection? Look how we did it for items

**Comment by Gorzen** [comment]
As I said in one of my review messages, I don't think we should have a courses collection on firebase, it's really not needed.
Courses should be part on an enum, where each different values of the enum have a unique value. Then you just put the values of the courses the player is in, in the player fields of firebase and when you read from firebase you give him the proper courses. By looking at the values you stored for the user.

File: app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java (hunk 2 above)
Comment (JulesssG) [reply app/src/main/java/ch/epfl/sweng/studyup/MainActivity.java]: Sorry my mistake, I pushed a commit with, normally, no typos like this anymore
